### PR TITLE
feat: saved products 

### PR DESCRIPTION
**Pull request by KambojRajan**
The previous code had an issue with Tanstack's caching, which is now fixed.

Currently, I make a call for each product in the carousel to check if it's saved. This approach is acceptable for displaying the top 10 products.

Alternatively, I could use GSM to fetch saved products once and update a  GS without extra queries. However, this adds client-side overhead with useEffect and other hook.

### CODE REVIEW

File: app/components/Navbar/SavedItems.tsx [hunk 1]
```diff
@@ -1,10 +1,132 @@
...
+import { Button } from "../ui/button";
+import Image from "next/image";
+
+import Link from "next/link";
+import { cn } from "@/app/utils";
+import { CategoryEnum, ConditionEnum } from "@/types";
+
+import DeleteSavedProducts from "./delete-saved-products";
+import { auth } from "@/auth";
+import { fetchSavedProduct, getAllSaved } from "@/lib/actions/products.actions";
```
Comment (heydoyouknowme0) [inline app/components/Navbar/SavedItems.tsx:20]: getAllSaved I think commit nahi hua

**Comment by som-04** [comment]
image

unable to save most of the items

File: app/components/Navbar/SavedItems.tsx [hunk 2]
```diff
@@ -1,10 +1,96 @@
...
+        <h1 className="text-2xl font-bold">Saved Products</h1>
+        <p className="text-muted-foreground">Manage your saved products</p>
+      </header>
+      <ul
+        className={cn(
+          "h-[calc(100dvh-7rem)] overflow-y-auto",
+          "flex flex-col space-y-4",
+        )}
+      >
+        {Array.from(savedProducts?.entries() ?? []).map(([id, product]) => (
```
Comment (heydoyouknowme0) [inline app/components/Navbar/SavedItems.tsx:36]:
```suggestion
        {(...savedProducts?.entries() ?? []).map((id, product) => (
```

Comment (KambojRajan) [reply app/components/Navbar/SavedItems.tsx:36]: but its the same thing. then what changes by doing that?

Comment (heydoyouknowme0) [reply app/components/Navbar/SavedItems.tsx:36]: Compiler

**Pull request closed without merging**